extensions/amp-story-subscriptions/0.1/test/test-amp-story-subscriptions.js

```diff
@@ -25,51 +28,169 @@ describes.realWin(
 
+      const platformConfig = {
+        'services': [
+          {
+            'authorizationUrl':
+              'https://scenic-2017.appspot.com/amp-entitlements?pubid=scenic-2017.appspot.com&meter=1&rid=READER_ID',
+            'pingbackUrl':
+              'https://scenic-2017.appspot.com/amp-pingback?pubid=scenic-2017.appspot.com',
+            'actions': {
+              'login': 'https://scenic-2017.appspot.com/signin?rid=READER_ID',
+              'subscribe': 'https://scenic-2017.appspot.com/subscribe',
+            },
+          },
+          {
+            'serviceId': 'subscribe.google.com',
+          },
+        ],
+      };
+      doc.head.appendChild(
+        <script type="application/json" id="amp-subscriptions">
+          {JSON.stringify(platformConfig)}
+        </script>
+      );
+
+      const pageConfig = {
+        '@context': 'http://schema.org',
+        '@type': 'NewsArticle',
+        'isAccessibleForFree': 'False',
+        'isPartOf': {
+          '@type': ['CreativeWork', 'Product'],
+          'name': 'Product A',
+          'productID': 'scenic-2017.appspot.com:news',
+        },
+      };
+      doc.head.appendChild(
+        <script type="application/ld+json">{JSON.stringify(pageConfig)}</script>
+      );
+
+      subscriptionService = new SubscriptionService(env.ampdoc);
+      subscriptionService.start();
+      env.sandbox
+        .stub(Services, 'subscriptionsServiceForDoc')
+        .returns(Promise.resolve(subscriptionService));
+
+      const storyEl = doc.createElement('amp-story');
       subscriptionsEl = (
         <amp-story-subscriptions layout="container"> </amp-story-subscriptions>
       );
-      const storyEl = doc.createElement('amp-story');
-      storyEl.appendChild(subscriptionsEl);
-      doc.body.appendChild(storyEl);
-
       storySubscriptions = new AmpStorySubscriptions(subscriptionsEl);
+      storyEl.appendChild(subscriptionsEl);
+      win.document.body.appendChild(storyEl);
+      await subscriptionsEl.whenBuilt();
     });
 
-    it('should contain amp-subscriptions attributes', async () => {
-      await subscriptionsEl.whenBuilt();
+    it('should contain amp-subscriptions attributes: subscriptions-dialog', async () => {
       expect(
         subscriptionsEl
           .querySelector('div')
           .hasAttribute('subscriptions-dialog')
       ).to.equal(true);
+    });
+
+    it('should contain amp-subscriptions attributes: subscriptions-display', async () => {
       expect(
         subscriptionsEl
           .querySelector('div')
           .getAttribute('subscriptions-display')
       ).to.equal('NOT granted');
     });
 
-    it('should display the blocking paywall on state update', async () => {
-      await storySubscriptions.buildCallback();
+    describe('should activate subscription platform and show paywall on dialog UI state update to true', async () => {
+      const maybeRenderDialogForSelectedPlatformSpy = env.sandbox.spy(
```

Comment:
Please try to wrap all code in before each so that it gets cleaned up between tests. Otherwise you get weird bugs where you are leaking state from one test to another.